tests/test_crud_data.py

```diff
@@ -0,0 +1,373 @@
+from core_lib.error_handling.status_code_exception import StatusCodeException
+from tests.test_data.test_utils import connect_to_mem_db
+from core_lib.rule_validator.rule_validator import ValueRuleValidator, RuleValidator
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+
+
+class Data(Base):
+    __tablename__ = 'crud_test'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False, default="")
+
+
+class CRUDInit(CRUDDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool)
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, Data, connect_to_mem_db(), CRUDInit.rules_validator)
+
+
+class DataSoftDelete(Base, SoftDeleteMixin):
+    __tablename__ = 'crud_softdel_test'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False, default="")
+
+
+class CRUDSoftDeleteInit(CRUDSoftDeleteDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool)
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, DataSoftDelete, connect_to_mem_db(), CRUDSoftDeleteInit.rules_validator)
+
+
+class DataSoftDeleteToken(Base, SoftDeleteMixin, SoftDeleteTokenMixin):
+    __tablename__ = 'crud_softdel_token_test'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False, default="")
+
+
+class CRUDSoftDeleteTokenInit(CRUDSoftDeleteWithTokenDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool)
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, DataSoftDeleteToken, connect_to_mem_db(), CRUDSoftDeleteTokenInit.rules_validator)
+
+
+class TestCrud(unittest.TestCase):
+
+    def test_crud(self):
+        crud = CRUDInit()
+
+        crud.create({'name': 'test_name', 'email': 'abc@def.com', 'active': True})
+
+        data = result_to_dict(crud.get(1))
+        self.assertNotEqual(data, None)
+        self.assertDictEqual(data, {'id': 1, 'name': 'test_name', 'email': 'abc@def.com', 'active': True})
+        self.assertEqual(data['id'], 1)

```

Comment:
this is redundant. above the assertDict is already validation the values.  

the assertDict is better in a way that it will also cover new properties